src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -2983,155 +2983,246 @@ int SketchObject::split(int GeoId, const Base::Vector3d &point)
-                transferConstraints(GeoId, PointPos::mid, newIds[0], PointPos::mid);
-                transferConstraints(GeoId, PointPos::end, newIds[1], PointPos::end, true);
-                ok = true;
-            }
+                transferConstraints(GeoId, PointPos::start, newId0, PointPos::start, true);
+                transferConstraints(GeoId, PointPos::mid, newId0, PointPos::mid);
+                transferConstraints(GeoId, PointPos::end, newId1, PointPos::end, true);
+            });
+    }
+    else if (geo->isDerivedFrom(Part::GeomArcOfConic::getClassTypeId())) {
+        ok = createGeosFromNonPeriodic(
+            static_cast<const Part::GeomBoundedCurve *>(geo),
+            [](const Part::GeomCurve* curve, double startParam, double endParam) {
+                auto newArc = static_cast<Part::GeomArcOfConic *>(curve->clone());
+                newArc->setRange(startParam, endParam);
+                return newArc;
+            },
+            [this, &newConstraints](int GeoId, int newId0, int newId1) {
+                // apply appropriate constraints on the new points at split point
+                Constraint* joint = new Constraint();
+                joint->Type = Coincident;
+                joint->First = newId0;
+                joint->FirstPos = PointPos::end;
+                joint->Second = newId1;
+                joint->SecondPos = PointPos::start;
+                newConstraints.push_back(joint);
+
+                // TODO: Do we apply constraints on center etc of the conics?
+
+                // transfer constraints from start and end of original
+                transferConstraints(GeoId, PointPos::start, newId0, PointPos::start, true);
+                transferConstraints(GeoId, PointPos::end, newId1, PointPos::end, true);
+            });
+    }
+    else if (geo->getTypeId() == Part::GeomBSplineCurve::getClassTypeId()) {
+        const Part::GeomBSplineCurve *bsp = static_cast<const Part::GeomBSplineCurve *>(geo);
+
+        // what to do for periodic b-splines?
+        if (bsp->isPeriodic()) {
+            ok = createGeosFromPeriodic(
+                static_cast<const Part::GeomCurve *>(geo),
+                [](const Part::GeomCurve* curve, double startParam, double endParam) {
+                    auto newBsp = static_cast<Part::GeomBSplineCurve *>(curve->clone());
+                    newBsp->Trim(startParam, endParam);
+                    return newBsp;
+                },
+                [](int, int) {
+                    // no constraints to transfer here, and we assume the split is to "break" the b-spline
+                });
+        }
+        else {
+            ok = createGeosFromNonPeriodic(
+                static_cast<const Part::GeomBoundedCurve *>(geo),
+                [](const Part::GeomCurve* curve, double startParam, double endParam) {
+                    auto newBsp = static_cast<Part::GeomBSplineCurve *>(curve->clone());

```

Comment:
Notes to self:
1. Who takes care of the lifetime of this just cloned pointer, specially if Trim exists due to an exception? (this applies to all geometry types, not just this one).
2. It is possible to pass here to trim parameters start=1 and end=1 (see calling code in lambda), client code should probably not allow this.

Reply:
Note to self:

For testing:
App.getDocument('hang').getObject('Sketch').addGeometry(Part.BSplineCurve([App.Vector(-50,60),App.Vector(-60,70),App.Vector(-70,80),App.Vector(-80,90),App.Vector(-90,100)],None,None,False,3,None,False),False)

ActiveSketch.solve()

App.getDocument('hang').getObject('Sketch').split(10,App.Vector(-89.9999999999999999999999999,99.9999999999999999999999999,0))